modules/contracts/src.ts/services/ethReader.ts

```diff
@@ -453,7 +454,12 @@ export class EthereumChainReader implements IVectorChainReader {
       return Result.fail(new ChainError(ChainError.reasons.ProviderNotFound));
     }
     try {
-      const gasPrice = await provider.getGasPrice();
+      let gasPrice = await provider.getGasPrice();
```

Comment:
Put this in the else so that you don't need to call the provider function if you're using gasnow.